src/components/cart.js

```diff
@@ -220,21 +237,54 @@ export default class Cart extends Component {
    */
   setQuantity(target, fn) {
     const id = target.getAttribute('data-line-item-id');
-    const item = this.model.lineItems.filter((lineItem) => lineItem.id === id)[0];
+    const item = this.model.lineItems.find((lineItem) => lineItem.id === id);
     const newQty = fn(item.quantity);
     return this.props.tracker.trackMethod(this.updateItem.bind(this), 'Update Cart', this.cartItemTrackingInfo(item, newQty))(id, newQty);
   }
 
   /**
+   * set cache using line items.
+   * @param {Array} lineItems - array of GraphModel line item objects.
+   */
+  updateCache(lineItems) {
+    const lineItemCache = this.lineItemCache.reduce((acc, item) => {
```

Comment:
maybe a different name than what you're referencing under `this`. I know this is the name I suggested, but it's confusing when both things are named the same but are objectively different.